libvast/src/system/indexer.cpp

```diff
@@ -25,12 +28,59 @@
 #include "vast/system/instrumentation.hpp"
 #include "vast/system/report.hpp"
 #include "vast/table_slice_column.hpp"
+#include "vast/value_index.hpp"
 #include "vast/view.hpp"
 
 #include <caf/attach_stream_sink.hpp>
 
 namespace vast::system {
 
+namespace v2 {
+
+caf::behavior indexer(caf::stateful_actor<indexer_state>* self, type index_type,
+                      caf::settings index_opts) {
+  self->state.name = "indexer" + to_string(index_type);
+  return {
+    [=](caf::stream<table_slice_column> in) {
+      VAST_DEBUG(self, "got a new table slice stream");
+      return caf::attach_stream_sink(
+        self, in,
+        [=](caf::unit_t&) {
+          self->state.idx = factory<value_index>::make(index_type, index_opts);
+          if (!self->state.idx) {
+            VAST_ERROR(self, "failed to construct index");
+            self->quit(); // TODO: choose right error code.
```

Comment:
```suggestion
            self->quit(ec::unspecified); // TODO: choose right error code.
```

Only marginally better. 🙂